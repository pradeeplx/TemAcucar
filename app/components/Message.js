import React from 'react'
import { View, Dimensions } from 'react-native'
import { fontFactor } from "../helpers"
import moment from 'moment'
import Colors from "../Colors"
import Sentence from "../components/Sentence"
import TimeAgo from "../components/TimeAgo"

export default Message = ({ message: { created_at, text }, fromCurrentUser, date }) => (
  <View style={{
  }}>
    { date && <View style={{
        marginBottom: 10,
        alignItems: 'center',
      }}>
        <Sentence>
          {date}
        </Sentence>
      </View>
    }
    <View style={{
      backgroundColor: (fromCurrentUser ? Colors.lightBeige : Colors.lightGray),
      padding: 10,
      borderRadius: 4,
      marginBottom: 10,
      width: (Dimensions.get('window').width * 0.75),
      alignSelf: (fromCurrentUser ? 'flex-end' : 'flex-start'),
    }}>
      <Sentence style={{
        color: Colors.darkGray,
      }}>
        {text}
      </Sentence>
      <Sentence style={{
        fontSize: 10 * fontFactor(),
        color: Colors.black,
        alignSelf: 'flex-end',
      }}>
        {moment(created_at).format('HH:mm')}
      </Sentence>
    </View>
  </View>
)
