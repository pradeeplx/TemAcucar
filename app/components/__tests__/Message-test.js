import 'react-native'
import React from 'react'
import Message from '../Message'
import renderer from 'react-test-renderer'

const props = {
  message: {
    created_at: 1482354068787,
    text: 'Foo',
  },
  date: 1482354242154,
  fromCurrentUser: false,
}

it('renders correctly', () => {
  const tree = renderer.create(
    <Message {...props} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders differently for current user', () => {
  const tree = renderer.create(
    <Message {...props} fromCurrentUser />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});


it('does not display date sentence when no date is given', () => {
  const tree = renderer.create(
    <Message {...props} date={null} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
