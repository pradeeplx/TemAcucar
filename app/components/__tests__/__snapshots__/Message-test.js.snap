exports[`test does not display date sentence when no date is given 1`] = `
<View
  style={Object {}}>
  <View
    style={
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "#EFEDED",
        "borderRadius": 4,
        "marginBottom": 10,
        "padding": 10,
        "width": 562.5,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "color": "#807F7F",
          },
        ]
      }>
      Foo
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "alignSelf": "flex-end",
            "color": "#000000",
            "fontSize": 15,
          },
        ]
      }>
      19:01
    </Text>
  </View>
</View>
`;

exports[`test renders correctly 1`] = `
<View
  style={Object {}}>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 10,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          undefined,
        ]
      }>
      HOJE
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "#EFEDED",
        "borderRadius": 4,
        "marginBottom": 10,
        "padding": 10,
        "width": 562.5,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "color": "#807F7F",
          },
        ]
      }>
      Foo
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "alignSelf": "flex-end",
            "color": "#000000",
            "fontSize": 15,
          },
        ]
      }>
      19:01
    </Text>
  </View>
</View>
`;

exports[`test renders differently for current user 1`] = `
<View
  style={Object {}}>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 10,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          undefined,
        ]
      }>
      HOJE
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "flex-end",
        "backgroundColor": "#FAF6F1",
        "borderRadius": 4,
        "marginBottom": 10,
        "padding": 10,
        "width": 562.5,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "color": "#807F7F",
          },
        ]
      }>
      Foo
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#654C50",
            "fontFamily": "Avenir",
            "fontSize": 18,
          },
          Object {
            "alignSelf": "flex-end",
            "color": "#000000",
            "fontSize": 15,
          },
        ]
      }>
      19:01
    </Text>
  </View>
</View>
`;
